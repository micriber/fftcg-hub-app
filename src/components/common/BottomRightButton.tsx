import React from 'react';
import {
  GestureResponderEvent,
  StyleSheet,
  TouchableOpacity,
  View,
} from 'react-native';
import AntIcon from 'react-native-vector-icons/AntDesign';

type Props = {
  iconName: string;
  onPress: (event: GestureResponderEvent) => void;
};

const BottomRightButton = ({iconName, onPress}: Props) => {
  return (
    <View style={styles.bottomView}>
      <TouchableOpacity style={styles.searchButton} onPress={onPress}>
        <AntIcon name={iconName} size={20} />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  bottomView: {
    width: '100%',
    height: 75,
    justifyContent: 'center',
    alignItems: 'flex-end',
    position: 'absolute',
    bottom: 20,
  },
  searchButton: {
    backgroundColor: '#00BCD4',
    borderRadius: 50,
    borderWidth: 1,
    borderColor: '#fff',
    width: 60,
    height: 60,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 20,
  },
});

export default BottomRightButton;
