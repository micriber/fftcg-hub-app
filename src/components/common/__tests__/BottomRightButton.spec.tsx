/**
 * @format
 */

import 'react-native';
import React from 'react';
import BottomRightButton from '../BottomRightButton';
import {render} from '@testing-library/react-native';

// More about testing in this repository:
// https://github.com/bamlab/react-native-testing/blob/master/src/pages/Home/__tests__/Home.test.tsx
describe('[Component] BottomRightButton', () => {
  it('renders correctly', async () => {
    const root = render(
      <BottomRightButton iconName="search1" onPress={() => {}} />,
    );

    expect(root.toJSON()).toMatchSnapshot();
  });
});
