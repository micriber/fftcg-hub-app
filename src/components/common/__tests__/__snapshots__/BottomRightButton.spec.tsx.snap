// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] BottomRightButton renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "bottom": 20,
      "height": 75,
      "justifyContent": "center",
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00BCD4",
        "borderColor": "#fff",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 60,
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
        "width": 60,
      }
    }
    testID="BottomRightButton"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î™°
    </Text>
  </View>
</View>
`;
